coverity-scan.sh

```diff
@@ -1,62 +1,167 @@
+    --enable-plugin-freeipmi \
+    --enable-plugin-cups \
+    --enable-backend-prometheus-remote-write \
+    ${NULL}
 
-[ -f netdata-coverity-analysis.tgz ] && rm netdata-coverity-analysis.tgz
+  # TODO: enable these plugins too
+  #	--enable-plugin-xenstat \
+  #	--enable-backend-kinesis \
+  #	--enable-backend-mongodb \
 
-autoreconf -ivf
-./configure --enable-plugin-nfacct --enable-plugin-freeipmi
-"${covbuild}" --dir cov-int make -j${cpus} || exit 1
+  progress "Analyzing netdata..."
+  run "${covbuild}" --dir cov-int make -j${cpus} || exit 1
 
-echo >&2 "Compressing data..."
-tar czvf netdata-coverity-analysis.tgz cov-int || exit 1
+  echo >&2 "Compressing analysis..."
+  run tar czvf netdata-coverity-analysis.tgz cov-int || exit 1
 
-echo >&2 "Sending analysis for version ${version} ..."
-COVERITY_SUBMIT_RESULT=$(curl --progress-bar --form token="${token}" \
-  --form email=${COVERITY_SCAN_SUBMIT_MAIL} \
-  --form file=@netdata-coverity-analysis.tgz \
-  --form version="${version}" \
-  --form description="netdata, real-time performance monitoring, done right." \
-  https://scan.coverity.com/builds?project=${REPOSITORY})
+  echo >&2 "Sending analysis to coverity for netdata version ${version} ..."
+  COVERITY_SUBMIT_RESULT=$(debugrun curl --progress-bar \
+    --form token="${token}" \
+    --form email=${email} \
+    --form file=@netdata-coverity-analysis.tgz \
+    --form version="${version}" \
+    --form description="netdata, monitor everything, in real-time." \
+    https://scan.coverity.com/builds?project=${repo})
 
-echo ${COVERITY_SUBMIT_RESULT} | grep -q -e 'Build successfully submitted' || echo >&2 "scan results were not pushed to coverity. Message was: ${COVERITY_SUBMIT_RESULT}"
+  echo ${COVERITY_SUBMIT_RESULT} | grep -q -e 'Build successfully submitted' || echo >&2 "scan results were not pushed to coverity. Message was: ${COVERITY_SUBMIT_RESULT}"
 
-echo >&2 "Coverity scan mechanism completed"
+  progress "Coverity scan completed"
+}
+
+installit() {
+  progress "Downloading coverity..."
+  cd /tmp || exit 1
+
+  [ -f "${COVERITY_BUILD_VERSION}.tar.gz" ] && run rm -f "${COVERITY_BUILD_VERSION}.tar.gz"
+  debugrun curl --remote-name --remote-header-name --show-error --location --data "token=${token}&project=${repo}" https://scan.coverity.com/download/linux64
+
+  if [ -f "${COVERITY_BUILD_VERSION}.tar.gz" ]; then
+    progress "Installing coverity..."
+    cd /opt || exit 1
+    run sudo tar -z -x -f  "/tmp/${COVERITY_BUILD_VERSION}.tar.gz" || exit 1
+    rm "/tmp/${COVERITY_BUILD_VERSION}.tar.gz"
+    export PATH=${PATH}:/opt/${COVERITY_BUILD_VERSION}/bin/
+  else
+    fatal "Failed to download coverity tool tarball!"
+  fi
+
+  # Validate the installation
+  covbuild="$(which cov-build 2>/dev/null || command -v cov-build 2>/dev/null)"
+  if [ -z "$covbuild" ]; then
+    fatal "Failed to install coverity."
+  fi
+
+  progress "Coverity scan tools are installed."
+  return 0
+}
+
+if [ "${1}" = "install" ]
```

Comment:
Would it make sense to use a `--with-install` option instead that will just run the install together with the run?